### ISSUE TITLE
`PopoverDisclosure` is missing `aria-haspopup` attribute

### ISSUE THREAD

**Issue by ivyrze** [ariakit/ariakit#2819]
### Current behavior

The underlying `<button>` doesn't have a `aria-haspopup`, which provides contextualization for screen reader users of it's expected action.

`<MenuButton>`, on the other hand, does include a `aria-haspopup="menu"`.

### Steps to reproduce the bug

1. Open sandbox: https://codesandbox.io/s/ariakit-popover-bug-report-vdyqh8
2. See console logs or inspect DOM

### Expected behavior

It should include a `aria-haspopup="dialog"` since the popover has `role="dialog"`.

### Workaround

https://github.com/ariakit/ariakit/issues/2819#issuecomment-1722443227

**Comment by diegohaz** [comment]
Good catch! Thank you for reporting, @ivyrze!

Before a fix is published, you can work around it by manually adding `aria-haspopup` to `PopoverDisclosure`:

```jsx
<PopoverDisclosure aria-haspopup="dialog">
```

**Comment by ivyrze** [comment]
Thank you! Appreciate all of your work on this project, it's quickly become one of my favorites!
